examples/common/webapi.c

```diff
@@ -22,6 +22,16 @@ void webapi_init(const webapi_desc_t* desc) {
     if (before_init_state.dbg_connect_requested && state.funcs.dbg_connect) {
         state.funcs.dbg_connect();
     }
+
+    /* Wrap _webapi_input so we don't need to sweat the string to char* conversion in JavaScript */
+    EM_ASM({
+        const saveWebInput = Module["_webapi_input"];
+        Module["_webapi_input"] = function(text) {
+            const utf8 = stringToNewUTF8(text);
+            saveWebInput(utf8);
+            Module["_webapi_free"](utf8);
+        }
+    });
```

Comment:
Hmm, "interesting" solution :)

I'll put an ifdef __EMSCRIPTEN__ around it though.

Reply:
@floooh thanks. heh. I couldn't believe the effort of doing this was pushed off to the "user" in JavaScript land.

Reply:
I think embind automatically takes care of such things, but this generates so much bloat that I prefer the 'manual' solution.